### PR TITLE
Add ensembl ids

### PR DESCRIPTION
**Pull request by KatalinaBobowik**
Add ENSEMBL IDs to expression output data

### CODE REVIEW

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py [hunk 1]
```diff
@@ -175,6 +177,13 @@ def create_struct(gene):
     data_summary = data_summary.rename({'index': 'gene_symbol'}, axis='columns')
     # add in cell type info
     data_summary['cell_type_id'] = celltype
+    # add in ENSEMBL IDs
+    gtf = pd.read_csv(GENCODE_GTF, sep='\t', header=None, index_col=False, skiprows=5)
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Would it be easier to use https://hail.is/docs/0.2/experimental/index.html#hail.experimental.import_gtf here instead?

Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Doesn't that get imported as a hail table? I only need a pandas df here

Comment (lgruen) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Yes, but you could easily convert the Hail Table to a Pandas DataFrame. You then wouldn't need any of that parsing code below.

Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Would you happen to know how to import the GTF? Using the same GTF file I specified at the top of the code and loading in using
```
hl.experimental.import_gtf(GENCODE_GTF, reference_genome='GRCh38', skip_invalid_contigs=True, force_bgz=True)
```
I get 
```
Error summary: ZipException: File does not conform to block gzip format.
```

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py [hunk 2]
```diff
@@ -28,6 +29,7 @@
 TOB_WGS = dataset_path('mt/v7.mt/')
 FREQ_TABLE = dataset_path('joint-calling/v7/variant_qc/frequencies.ht/', 'analysis')
 VEP_ANNOTATION = dataset_path('tob_wgs_vep/v1/vep105_GRCh38.mt/')
+GENCODE_GTF = 'gs://cpg-reference/gencode/gencode.v38.annotation.gtf.gz'
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Unrelated to this PR, but noting that we should likely rerun this with VEP 104?

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py (hunk 1 above)
Comment (lgruen) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: I think this is still pending on the discussion here: https://centrepopgen.slack.com/archives/C018KFBCR1C/p1661410430973899?thread_ts=1658208231.938539&cid=C018KFBCR1C

**Review by lgruen — CHANGES_REQUESTED** [review]

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py (hunk 2 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: I'm happy to rerun VEP using v104 if that's what's decided upon

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py (hunk 1 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: This works using `force = True`, and I've added this in now.

**Comment by KatalinaBobowik** [comment]
Just to confirm, have we decide on Gencode v 38? If so, I just need the correct link, as `gs://cpg-reference/gencode/gencode.v38.annotation.gtf.gz` currently doesn't exist

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py [hunk 3]
```diff
@@ -28,6 +28,7 @@
 TOB_WGS = dataset_path('mt/v7.mt/')
 FREQ_TABLE = dataset_path('joint-calling/v7/variant_qc/frequencies.ht/', 'analysis')
 VEP_ANNOTATION = dataset_path('tob_wgs_vep/v1/vep105_GRCh38.mt/')
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: This should point to `vep104`, right?

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py [hunk 4]
```diff
@@ -172,9 +173,12 @@ def create_struct(gene):
     data_summary = pd.DataFrame(
         log_cpm.apply(create_struct, axis=0), columns=['data']
     ).reset_index()
-    data_summary = data_summary.rename({'index': 'gene_symbol'}, axis='columns')
+    data_summary = data_summary.rename({'index': 'gene_name'}, axis='columns')
     # add in cell type info
     data_summary['cell_type_id'] = celltype
+    # add in ENSEMBL IDs
+    gtf = hl.experimental.import_gtf(GENCODE_GTF, force=True, reference_genome='GRCh38', skip_invalid_contigs=True)
```
Comment (lgruen) [inline scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Now that we have `.bgz`, you shouldn't need the `force=True` anymore.

**Review by lgruen — APPROVED** [review]

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py (hunk 4 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Ok, good to know. I've removed it now

File: scripts/eqtl_hail_batch/generate_eqtl_spearman.py (hunk 3 above)
Comment (KatalinaBobowik) [reply scripts/eqtl_hail_batch/generate_eqtl_spearman.py]: Yes, thanks for that. I've fixed the path

**Pull request merged**